### PR TITLE
EN 1140 attributes list table

### PR DESCRIPTION
**Pull request by mcarta-rt**

### CODE REVIEW

**Comment by entando-service** [comment]
All is well

File: src/state/data-types/actions.js [hunk 1]
```diff
@@ -23,7 +39,25 @@ export const fetchDataTypes = (page = { page: 1, pageSize: 10 }, params = '') =>
           dispatch(setDataTypes(json.payload));
           dispatch(toggleLoading('dataTypes'));
           dispatch(setPage(json.metaData));
-        } else if (json && json.errors) {
+        } else {
+          dispatch(addErrors(json.errors.map(err => err.message)));
+          dispatch(toggleLoading('dataTypes'));
```
Comment (alessioscalici) [inline src/state/data-types/actions.js]: Please remove the two `dispatch(toggleLoading('dataTypes'));` calls from the if/else branches, and make a single call outside the if/else

File: src/state/data-types/actions.js [hunk 2]
```diff
@@ -23,7 +39,25 @@ export const fetchDataTypes = (page = { page: 1, pageSize: 10 }, params = '') =>
...
+);
+
+export const fetchDataType = dataTypeCode => dispatch => (
+  new Promise((resolve) => {
+    dispatch(toggleLoading('dataTypes'));
+    getDataType(dataTypeCode).then((response) => {
+      response.json().then((json) => {
+        if (response.ok) {
+          dispatch(setDataTypes([json.payload]));
+          dispatch(initialize('dataType', json.payload));
```
Comment (alessioscalici) [inline src/state/data-types/actions.js]: Please call  `dispatch(toggleLoading('dataTypes'));` outside the if/else block

File: src/ui/common/attributes/AttributeListTable.js [hunk 3]
```diff
@@ -0,0 +1,87 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { FormattedMessage } from 'react-intl';
+// import { formattedText } from 'frontend-common-components';
```
Comment (alessioscalici) [inline src/ui/common/attributes/AttributeListTable.js]: please remove the comment

File: src/ui/data-types/add/AddDataTypesPage.js [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+            <BreadcrumbItem route={ROUTE_DATA_TYPE_LIST}>
+              <FormattedMessage id="menu.dataType" />
+            </BreadcrumbItem>
+            <BreadcrumbItem active>
+              <FormattedMessage id="app.add" />
+            </BreadcrumbItem>
+          </Breadcrumb>
+        </Col>
+      </Row>
+      <PageTitle
```
Comment (alessioscalici) [inline src/ui/data-types/add/AddDataTypesPage.js]: Please enclose PageTitle in a Row/Col block

File: src/ui/data-types/edit/EditDataTypesPage.js [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+            <BreadcrumbItem route={ROUTE_DATA_TYPE_LIST}>
+              <FormattedMessage id="menu.dataType" />
+            </BreadcrumbItem>
+            <BreadcrumbItem active>
+              <FormattedMessage id="app.edit" />
+            </BreadcrumbItem>
+          </Breadcrumb>
+        </Col>
+      </Row>
+      <PageTitle
```
Comment (alessioscalici) [inline src/ui/data-types/edit/EditDataTypesPage.js]: Please enclose PageTitle in a Row/Col block

File: src/ui/data-types/list/DataTypeListTable.js [hunk 6]
```diff
@@ -23,13 +23,15 @@ class DataTypeListTable extends Component {
...
     return this.props.datatypes.map(datatype => (
-      <tr key={datatype.code}>
+      <tr key={datatype.name}>
         <td className="DataTypeListRow__td">{datatype.name}</td>
         <td className="DataTypeListRow__td text-center">{datatype.code}</td>
         <td className="DataTypeListRow__td text-center">
           <DataTypeStatusIcon
             status={datatype.status}
-            title={formattedText(`dataType.table.status.${datatype.status}`)}
+            // title={formattedText(`dataType.table.status.${datatype.status}`)}
```
Comment (alessioscalici) [inline src/ui/data-types/list/DataTypeListTable.js]: Please remove this comment

File: test/state/data-types/actions.test.js [hunk 7]
```diff
@@ -1,69 +1,173 @@
...
+      });
+    });
+    describe('fetchDataTypeAttribute', () => {
+      it('fetchDataTypeAttribute calls setSelectedAttribute actions', (done) => {
+        getDataTypeAttribute.mockImplementation(mockApi({ payload: DATA_TYPE_ATTRIBUTE }));
+        store.dispatch(fetchDataTypeAttribute()).then(() => {
+          const actions = store.getActions();
+          expect(actions).toHaveLength(1);
+          expect(actions[0]).toHaveProperty('type', SET_SELECTED);
+          expect(actions[0]).toHaveProperty('payload.dataTypeAttributeCode');
```
Comment (alessioscalici) [inline test/state/data-types/actions.test.js]: This is redundant, the next expectation covers this as well

**Comment by entando-service** [comment]
All is well

**Comment by entando-service** [comment]
All is well

File: src/state/data-types/actions.js (hunk 2 above)
Comment (nicpuddu) [reply src/state/data-types/actions.js]: done

**Pull request merged**